test/integration/com/lucastex/grails/fileuploader/FileUploaderServiceSpec.groovy

```diff
@@ -0,0 +1,212 @@
+package com.lucastex.grails.fileuploader
+
+
+import org.springframework.context.MessageSource
+import org.springframework.context.support.ResourceBundleMessageSource
+import java.nio.file.FileSystems
+import java.nio.file.Path
+import org.codehaus.groovy.grails.commons.DefaultGrailsApplication
+import spock.lang.*
+import grails.test.spock.IntegrationSpec
+import com.lucastex.grails.fileuploader.FileUploaderService
+import com.lucastex.grails.fileuploader.UFileType
+
+class FileUploaderServiceSpec extends IntegrationSpec {
+
+    FileUploaderService fileUploaderService
+    def grailsApplication = new DefaultGrailsApplication()
+
+    String group
+    UFile ufileInstance
+
+    MessageSource getI18n() {
+        URL url = new File('grails-app/i18n').toURI().toURL()
+        MessageSource messageSource = new ResourceBundleMessageSource()
+        messageSource.bundleClassLoader = new URLClassLoader(url)
+        messageSource.basename = 'messages'
+        return messageSource
+    }
+
+    private void setupConfig() {
+        Map logoConfig = [
+            maxSize: 1024 * 1024 * 10,
+            allowedExtensions: ["jpg", "jpeg", "png"],
+            path: "./web-app/user-content/images/logo/",
+            container: "altruhelp_test"
+        ]
+        fileUploaderService.grailsApplication.config.fileuploader = [storageTypes: "", logo: logoConfig]
+    }
+
+    private File getTestFile(String fileName = "") {
+        // A destination temporary path to copy our test file, since after ufile upload the file gets deleted
+        fileName = fileName ?: "test-logo.png"
+        Path destination = FileSystems.getDefault().getPath(fileName)
+        Path source = FileSystems.getDefault().getPath("test", "integration", "test-files", "logo.png")
+
+        // Copying file to another location so that original file does not get deleted from repo.
+        File testFile
+        try {
+            java.nio.file.Files.copy(source, destination)
+        } catch (Exception e) {
+            log.warn "Exception occured while copying test file", e
+        }
+        testFile = new File("./${fileName}")
+
+        return testFile
+    }
+
+    def setup() {
+        fileUploaderService = new FileUploaderService()
```

Comment:
Delete all these manual code of instantiating service instance. Grails will do it for you by simply injecting it to test case as you have already done in at Line 16.